### ISSUE TITLE
Integration test `SchemaAgreementTest.should_unset_flag_on_failed_agreement` is flaky

### ISSUE THREAD

**Issue by fruch** [scylladb/java-driver#170]
Seem that on branch enterprise-2022.2 tests `SchemaAgreementTest.should_unset_flag_on_failed_agreement` keep 
on failing with like:

```
23:57:25  Starting com.datastax.driver.core.SchemaAgreementTest.should_unset_flag_on_failed_agreement [Test #317]...
23:57:25  FAILED : should_unset_flag_on_failed_agreement
23:57:25  Test   : 00:00:00
23:57:25  Elapsed: 01:09:02
23:57:25  
23:57:25      4141961 [cluster5588-reconnection-1] ERROR com.datastax.driver.core.ControlConnection - [Control connection] Cannot connect to any host, scheduling retry in 512000 milliseconds
23:57:25      4141972 [cluster5588-nio-worker-19] WARN  com.datastax.driver.core.Connection - Error creating netty channel to /127.0.1.2:38783
23:57:25  io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.1.2:38783
23:57:25  	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
23:57:25  	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:716)
23:57:25  	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:225)
23:57:25  	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
23:57:25  	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:634)
23:57:25  	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:581)
23:57:25  	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:498)
23:57:25  	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:460)
23:57:25  	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:131)
23:57:25  	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
23:57:25  	at java.lang.Thread.run(Thread.java:750)
23:57:25  Caused by: java.net.ConnectException: Connection refused
23:57:25  	... 11 more
23:57:30      4145646 [main] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm stop --config-dir=/tmp/1667511261457-0
23:57:30      4146268 [main] INFO  com.datastax.driver.core.CCMBridge - Error during tests, kept C* logs in /tmp/1667511261457-0
23:57:30  [ERROR] Tests run: 326, Failures: 1, Errors: 0, Skipped: 12, Time elapsed: 4,148.029 s <<< FAILURE! - in TestSuite
23:57:30  [ERROR] should_unset_flag_on_failed_agreement(com.datastax.driver.core.SchemaAgreementTest)  Time elapsed: 0.01 s  <<< FAILURE!
23:57:30  com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
23:57:30  	at com.datastax.driver.core.SchemaAgreementTest.should_unset_flag_on_failed_agreement(SchemaAgreementTest.java:55)
23:57:30  Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
23:57:30  	at com.datastax.driver.core.SchemaAgreementTest.should_unset_flag_on_failed_agreement(SchemaAgreementTest.java:55)
23:57:30  
23:57:30      4146330 [Thread-76] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm stop --config-dir=/tmp/1667511790109-0
23:57:30      4146331 [Thread-80] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm stop --config-dir=/tmp/1667512049589-0
23:57:33      4149960 [Thread-80] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm remove --config-dir=/tmp/1667512049589-0
23:57:33      4149965 [Thread-76] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm remove --config-dir=/tmp/1667511790109-0
23:57:36  [INFO] 
23:57:36  [INFO] Results:
23:57:36  [INFO] 
23:57:36  [ERROR] Failures: 
23:57:36  [ERROR]   SchemaAgreementTest.should_unset_flag_on_failed_agreement:55 ? NoHostAvailable
23:57:36  [INFO] 
23:57:36  [ERROR] Tests run: 323, Failures: 1, Errors: 0, Skipped: 9
```

https://jenkins.scylladb.com/job/enterprise-2022.2/job/driver-tests/job/java-driver-matrix-test/8/console
https://jenkins.scylladb.com/job/enterprise-2022.2/job/driver-tests/job/java-driver-matrix-test/7/console
https://jenkins.scylladb.com/job/enterprise-2022.2/job/driver-tests/job/java-driver-matrix-test/6/console

seems like cluster with node 127.0.1.2:38783, is being used in some tests before, and it's not in a health state

this test seems to be working in master and 5.1 just fine:
https://jenkins.scylladb.com/job/scylla-master/job/driver-tests/job/java-driver-matrix-test/894/testReport/3.7.1-scylla-2.com.datastax.driver.core/SchemaAgreementTest/should_unset_flag_on_failed_agreement/history/
 https://jenkins.scylladb.com/view/scylla-5.1/job/scylla-5.1/job/driver-tests/job/java-driver-matrix-test/7/console

**Comment by avelanarius** [comment]
Had trouble reproducing this both locally and on GitHub Actions, however I saw two occurances of this: https://github.com/scylladb/java-driver/actions/runs/3531706886/jobs/5925177185 and https://github.com/scylladb/java-driver/actions/runs/3531706886/jobs/5925176794

Unfortunately right now I don't have any working theory on what causes this problem.

**Comment by avelanarius** [comment]
Yesterday magically the test started to fail consistently for me on GitHub Actions (since this is a flaky problem, probably the timings on GitHub Actions changed for some reason).

The first thing I tried was only running the offending test class - `SchemaAgreementTest`, but running only this single test worked correctly.

Next, I turned on different debugging info (`TRACE` level in https://github.com/scylladb/java-driver/blob/scylla-3.x/driver-core/src/test/resources/log4j.properties) and experimented with adding additional logging.

Based on that: `SchemaAgreementTest` has multiple test methods. Most of them are executed sequentially, but `should_unset_flag_on_failed_agreement` is executed much later (multiple minutes after all previous methods in `SchemaAgreementTest`). In the meantime, tests from other classes are executed.

When finally `should_unset_flag_on_failed_agreement` is executed, the session created minutes earlier for previous tests from `SchemaAgreementTest` is broken. From `TRACE` logs I observed that the session has received a "DOWN" event (not sure if it's a CQL event). 

So I'll now continue figuring out:
1. Why `should_unset_flag_on_failed_agreement` is executed so much later.
2. Why session gets broken - session staying idle for few minutes shouldn't cause problems. Has the cluster stopped in the meantime?

**Comment by avelanarius** [comment]
By the way, the test failed on other Scylla versions and on Cassandra, so this is a purely driver issue (not only 2022.2).

**Comment by avelanarius** [comment]
Found the bug. 

In order to find the issue, I had to get all the logs of a `Cluster`/`Session` that was involved in the error message:
```
com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
```

I added a print that before each `execute()` would print the name of cluster the query was executed against. Thanks to this, I could find all the logs of this exact `Cluster`/`Session` (involved in failed test). Looking at those logs, I saw that the `Session` received `DOWN` CQL event. Then I looked whether the node was actually stopped - it turned out an unrelated test executed `ccm stop`. The test forgot to set a `dirtiesContext = true` flag.
